### PR TITLE
CI setup

### PR DESCRIPTION
**Pull request by jasonpaige**
### JIRA link (if applicable) ###
Part of RPE-40

### Change description ###
Setup travis and codacy integration. Can't test coverage until this is in master :(
Need to reintegrate using jenkins user once I get my vault policy sorted

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+before_install:
+  - sudo apt-get install jq
+  - wget -O ~/codacy-coverage-reporter-assembly-latest.jar $(curl https://api.github.com/repos/codacy/codacy-coverage-reporter/releases/latest | jq -r .assets[0].browser_download_url)
+  - docker build -t "draft-store:pgsql" ./docker/database/
+  - docker run -d -e DRAFT_STORE_DB_PASSWORD=$DRAFT_STORE_DB_PASSWORD -p $DRAFT_STORE_DB_PORT:5432 draft-store:pgsql
```
Comment (timja-kainos) [inline .travis.yml]: is service2service auth not also needed for functional tests?

Comment (jasonpaige) [reply .travis.yml]: seems to pass to im assuming not? :|

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+before_install:
+  - sudo apt-get install jq
+  - wget -O ~/codacy-coverage-reporter-assembly-latest.jar $(curl https://api.github.com/repos/codacy/codacy-coverage-reporter/releases/latest | jq -r .assets[0].browser_download_url)
+  - docker build -t "draft-store:pgsql" ./docker/database/
```
Comment (timja-kainos) [inline .travis.yml]: is docker-compose not available on travis?

File: .travis.yml (hunk 1 above)
Comment (timja-kainos) [reply .travis.yml]: @rkondratowicz should this be happening?

File: .travis.yml (hunk 2 above)
Comment (jasonpaige) [reply .travis.yml]: it is but I think there are challenges then with pulling out artefacts like coverage reports. This way it's just the db inside a container and it's super simples to grab the bits I need to ship to codacy

**Pull request merged**